### DISCUSSION TITLE
n-dimensional integration using `bayesquad`

### DISCUSSION THREAD

**Discussion by gerdm** [probabilistic-numerics/probnum#512] [Q&A] [answered]
Hi,

I've been trying to use `probnum`'s  `bayesquad` function to integrate a vector-valued function. However, I cannot get it to work. Here's a small example

```python
from probnum.quad import bayesquad

def f(x):
    return x.T @ x

domain=jnp.array([-3, 3])
bayesquad(f, 2, domain=domain)
```

but I get the following error

```python
ValueError: matmul: Input operand 1 has a mismatch in its core dimension 0, with gufunc signature (n?,k),(k,m?)->(n?,m?) (size 2 is different from 50)
```

I've tried different different domain configurations, but they do not work either.

```python
domain = np.array([
[-3, 3],
[-3, 3]
])

domain = np.array([-3, 3, -3, 3])
```

How can I compute this integral?

**Comment by mmahsereci** [comment, marked as answer]
hi @gerdm , I moved the discussion to an issue for better visibility. It's here #513 .
